### PR TITLE
sbt 1.0 support

### PR DESCRIPTION
**Pull request by LolHens**
I replaced all the deprecated operators and changed this to be an autoPlugin and now it compiles with sbt 0.13.16 and sbt 1.0.0
Because of these changes the pr is not compatible with sbt 0.12 anymore.
The tests all pass, and everything seems to work fine, but I am still new to sbt plugin development and have not yet submitted many prs so feedback is appreciated if things are done improperly.

### CODE REVIEW

File: src/sbt-test/proguard/akka/project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version=0.12.4
+sbt.version=0.13.16
```
Comment (eed3si9n) [inline src/sbt-test/proguard/akka/project/build.properties]: I suggest removing `build.properties` from scripted tests.

File: src/main/scala/com/typesafe/sbt/ProguardKeys.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import com.typesafe.sbt.SbtProguard.autoImport.ProguardOptions.Filtered
+import com.typesafe.sbt.proguard.Merge
+import com.typesafe.sbt.proguard.Merge.Strategy
+import sbt.{File, SettingKey, TaskKey}
+
+trait ProguardKeys {
+  val proguardVersion = SettingKey[String]("proguard-version")
+  val proguardDirectory = SettingKey[File]("proguard-directory")
+  val proguardConfiguration = SettingKey[File]("proguard-configuration")
+  val binaryDeps = TaskKey[Seq[File]]("binaryDeps")
```
Comment (eed3si9n) [inline src/main/scala/com/typesafe/sbt/ProguardKeys.scala]: The key should have `proguard` prefixed, as `proguardBinaryDeps`, and use lower cased `taskKey[Seq[File]]("...")`.
http://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html#Avoid+namespace+clashes

The same for rest of the keys.

File: src/main/scala/com/typesafe/sbt/SbtProguard.scala [hunk 3]
```diff
@@ -1,180 +1,116 @@
...
 import com.typesafe.sbt.proguard.Merge
+import com.typesafe.sbt.proguard.Sbt10Compat._
+import sbt.Keys._
+import sbt.{Def, _}
 
-object SbtProguard extends Plugin {
+import scala.sys.process.Process
 
-  val Proguard = config("proguard").hide
+object SbtProguard extends AutoPlugin {
```
Comment (eed3si9n) [inline src/main/scala/com/typesafe/sbt/SbtProguard.scala]: AutoPlugin that uses built-in key should include:

```scala
override def requires = plugins.JvmPlugin
```

File: src/sbt-test/proguard/akka/project/SampleBuild.scala [hunk 4]
```diff
@@ -1,23 +1,25 @@
 import sbt._
 import sbt.Keys._
-import com.typesafe.sbt.SbtProguard._
+import com.typesafe.sbt.SbtProguard.autoImport._
+import com.typesafe.sbt.SbtProguard
 
 object SampleBuild extends Build {
```
Comment (eed3si9n) [inline src/sbt-test/proguard/akka/project/SampleBuild.scala]: `Build` trait won't work for sbt 1, if we want to run scripted test on it.

**Comment by LolHens** [comment]
Nice!
So this is ready for merge now?

**Comment by eed3si9n** [comment]
I just did a drive-by review, but we should get ok from @pvlugter.

**Comment by alexarchambault** [comment]
Would a release including that be underway? I'll use it straightaway in https://github.com/coursier/coursier/pull/629.

**Comment by wsxarcher** [comment]
A release would be awesome.

**Comment by pvlugter** [comment]
New release:

```
addSbtPlugin("com.lightbend.sbt" % "sbt-proguard" % "0.3.0")
```

**Pull request merged**